asapdiscovery-docking/asapdiscovery/docking/openeye.py

```diff
@@ -298,45 +316,104 @@ def _dock(
-                                posed_ligand=posed_ligand,
-                                probability=prob,
-                                provenance=self.provenance(),
                             )
-                            if return_for_disk_backend:
-                                docking_results.append(docked_result_json_path)
-                            else:
-                                docking_results.append(docking_result)
-                            if output_dir is not None:
-                                docking_result.write_docking_files(output_dir)
+                        else:
+                            input_pairs.append(set)
+
+                    # Create Docking Results Objects
+                    docking_results_objects = []
+                    if self.num_poses == 1:
+                        # this is simple for single pose results
+                        for input_pair, posed_ligand in zip(input_pairs, posed_ligands):
+                            docking_results_objects.append(
+                                POSITDockingResults(
+                                    input_pair=input_pair,
+                                    posed_ligand=posed_ligand,
+                                    probability=posed_ligand.tags[
+                                        DockingResultCols.DOCKING_CONFIDENCE_POSIT.value
+                                    ],
+                                    provenance=self.provenance(),
+                                )
+                            )
+                    else:
+                        # for multi-pose results, we need to split the results by input pair
+                        # create hashable dict of input pairs
+                        input_pair_dict = {
+                            input_pair.unique_name: input_pair
+                            for input_pair in input_pairs
+                        }
+
+                        # split results by input pair
+                        from collections import defaultdict
+
+                        results_dict = defaultdict(list)
+                        for input_pair, posed_ligand in zip(input_pairs, posed_ligands):
+                            results_dict[input_pair.unique_name].append(posed_ligand)
+
+                        # return results split by input pair
+                        for input_pair_name, posed_ligands in results_dict.items():
+                            docking_results_objects.append(
+                                POSITDockingResults(
+                                    input_pair=input_pair_dict[input_pair_name],
+                                    posed_ligand=Ligand.from_single_conformers(
+                                        posed_ligands
+                                    ),
+                                    provenance=self.provenance(),
+                                    probability=(
+                                        posed_ligands[0].tags[
```

Comment:
Can't they have different probabilities or am I missing something?

Reply:
The way I've written it, the DockingResults class gets a probability & pose id if the number of poses = 1, otherwise it sets it to None.